mongodb/pom.xml

```diff
@@ -112,6 +112,12 @@
             <artifactId>jboss-transaction-api_1.2_spec</artifactId>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>de.flapdoodle.embed</groupId>
+            <artifactId>de.flapdoodle.embed.mongo</artifactId>
+            <version>2.0.3</version>
+            <scope>test</scope>
+        </dependency>
```

Comment:
Even if the dependency is used just in mongoDb module, we usually manage the version in parent pom. For instance take a look to:
```xml
<dependency>
  <groupId>org.skyscreamer</groupId>
  <artifactId>jsonassert</artifactId>
  <version>${version.org.skyscreamer.jsonassert}</version>
  <scope>test</scope>
</dependency>
```

Reply:
TBH, I'm a bit surprised we need this? Don't we already start a MongoDB server when we run the tests? Why would we need another one?

Reply:
@fax4ever , thank you

Reply:
@gsmet , I think it's a little bit more convenient if we can run the tests directly from IDE. WDYT?

Reply:
Ah yes, usually, I start a MongoDB server for that.

If we want to do that, we should be consistent and do it everywhere. It's weird to have only one test working like that.

@DavideD I suppose it's you who should decide what you prefer for the test infrastructure.

Reply:
I agree with @gsmet on this: if we use this approach for one test we should try to be consistent.
The problem that I see with this PR is that II wouldn't expect an issue about adding a specific test to also change the way we run the tests in the project. It seems a task big enough to deserve a separate issue. 

Could you update this PR for OGM-1222 (by removing the mongodb server runner) and create new JIRA and PR with the MongoDBserverRunner that also updates the existing tests?

@ntviet18, Would you find it surprising to be able to run this test without having a server running but not the others?

I also noticed that we have  a HotRodServerLifeCycle in the infinispan-remote project . It is bound to the surefire plugin, do we need something similar here as well?

Thanks